context.go

```diff
@@ -33,10 +33,8 @@ func (nc *Conn) RequestWithContext(ctx context.Context, subj string, data []byte
 	}
 	// Check whether the context is done already before making
 	// the request.
-	select {
-	case <-ctx.Done():
+	if ctx.Err() != nil {
```

Comment:
Can this condition change after this call but before we evaluate ctx->Done() below? Leading to the same problem as before but with a slimmer chance of happening? /cc @kozlovic

Reply:
Sounds good.